src/WebJobs.Extensions.Http/HttpTriggerAttributeBindingProvider.cs

```diff
@@ -403,10 +403,11 @@ private static object ConvertValueIfNecessary(object value, Type targetType)
                 {
                     var underlyingTargetType = Nullable.GetUnderlyingType(targetType) ?? targetType;
 
-                    var jObject = value as JObject;
-                    if (jObject != null)
+                    var jToken = value as JToken;
+                    if (jToken != null)
                     {
-                        value = jObject.ToObject(targetType);
+                        // value is JObject or JArray
```

Comment:
It's a JToken - how do you know its an array or object? It could be a string, or scalar value right?

Reply:
Fixed

Reply:
comment is still wrong

Reply:
This still doesn't seem right. Here you know its not a JObject, but a JArray, right?